packages/opentelemetry/lib/tracing/tracerProvider.js

```diff
@@ -0,0 +1,150 @@
+  }
+
+  /**
+   * adds exporter type and config to currentConfig
+   * @param {exporter} exporter  exporter object with exporterType and config
+   */
+  addExporter (exporterObject, exporter) {
+    this.exporters[exporter.exporterType] = exporterObject;
+    this._currentConfig.exporters.push(exporter);
+  }
+
+  /**
+   * generates a span processor for exporter object containing a type and config
+   * @param {Object} exporter  exporter object with exporterType and config
+   */
+  registerExporter (exporter) {
+    const exporterObject = buildExporter(exporter.exporterType, exporter.config);
+    const spanProcessor = getBatchSpanProcessor(exporterObject);
+    this.addSpanProcessor(spanProcessor);
+    this.addExporter(exporterObject, exporter);
+  }
+
+  /**
+   * adds a spanprocessor object to the current active (default) TracerProvider
+   * @param {SpanProcessor} spanProcessor  exporter object with exporterType and config
+   */
+  addSpanProcessor (spanProcessor) {
+    this.provider.addSpanProcessor(spanProcessor);
+  }
+
+  /**
+   * adds an instrumentation object to the current active (default) TracerProvider
+   * @param {[InstrumentationOption]} instrumentationInstance  instrumentation object
+   */
+  registerInstrumentation (instrumentationInstance) {
+    registerInstrumentations({
+      instrumentation: [instrumentationInstance],
+      provider: this.provider
+    });
+  }
+
+  /**
+   * disables existing exporter if active
+   * @param {string} exporterType exporter types from AVAILABLE_EXPORTERS
+   */
+  async disableExporter (exporterType) {
+    if (this.exporters[exporterType]) {
+      await this.exporters[exporterType].shutdown();
+      delete this.exporters[exporterType];
+    }
+    this._currentConfig.exporters = this._currentConfig.exporters.filter(function (item) {
+      return item.exporterType !== exporterType;
+    });
+  }
+
+  /**
+   * disables the current tracer provider and all its associated exporters
+   */
+  async shutdown () {
+    this._currentConfig.active = false;
+    const exporterList = Object.keys(this.exporters);
+    await this.disableAllExporters();
+  }
+
+  /**
+   * checks if the current tracer provider is active
+   */
+  isAlive () {
+    return this.currentConfig.active;
+  }
+
+  /**
+   * registers the current tracer provider as the active tracer provider with the api
+   */
+  initializeTracer () {
+    this.provider.register();
+  }
+
+  /**
+   * resets the current active tracer provider to its original state
+   */
+  async reset () {
+    await this.disableAllExporters();
+    this.registerExporter({
+      exporterType: AVAILABLE_EXPORTERS.CONSOLE
+    });
+    this._currentConfig.active = true;
+  }
+
+  /**
+   * disables all exporters associated with the current active tracer provider
+   */
+  async disableAllExporters () {
+    const exporterList = Object.keys(this.exporters);
+    await forEachPromise(exporterList, async (exporterType) => {
```

Comment:
I usually use bluebird for such purpose:

```
import B from 'bluebird';

const exporterTypes = Object.keys(this.exporters);
await B.all(exporterTypes.map((exporterType) => this.disableExporter(exporterType)))
```

Reply:
yeah didnt want to rely on 3rd party lib for such a smaller use case.

Reply:
we anyway use bluebird evrywhere same as lodash, so it is safe to use these as they will be anyway present in node_modules